### PR TITLE
Add IntelliSense configuration property: compilerArgs

### PR DESCRIPTION
**Pull request by michelleangela**
- Add `compilerArgs` property to IntelliSense configuration
- When sending configuration to language client, separate any args from `compilerPath` and add to `compilerArgs`. The language client from now on expects `compilerPath` to only have path, and args will be in `compilerArgs`.
- Refactor `getBuildTasks` function to support `compilerArgs`
- Update cpp-tools API reference version to 3.0.0

### CODE REVIEW

File: Extension/package.json [hunk 1]
```diff
@@ -1505,7 +1517,7 @@
     "mkdirp": "^0.5.1",
     "plist": "^2.0.1",
     "tmp": "^0.1.0",
-    "vscode-cpptools": "^2.1.2",
+    "vscode-cpptools": "file:../../vscode-cpptools-api",
```
Comment (michelleangela) [inline Extension/package.json]: TODO: change API version to 2.2.0 when it is published

File: Extension/package.json [hunk 2]
```diff
@@ -394,6 +394,18 @@
...
+        "C_Cpp.default.compilerArgs": {
+          "type": [
+            "array",
+            "null"
+          ],
+          "items": {
+            "type": "string"
+          },
+          "default": null,
+          "description": "The value to use in configuration if \"compilerArgs\" is not specified.",
```
Comment (bobbrow) [inline Extension/package.json]: "either not specified or set to ${default}"

File: Extension/src/LanguageServer/configurations.ts [hunk 3]
```diff
@@ -302,6 +305,9 @@ export class CppProperties {
             // don't set a default when compileCommands is in use.
             configuration.compilerPath = this.defaultCompilerPath;
         }
+        if (isUnset(settings.defaultCompilerArgs)) { // Should this also have the other validations above?
```
Comment (bobbrow) [inline Extension/src/LanguageServer/configurations.ts]: Yes, I believe you should add the same validations as `compilerPath`.

**Review by bobbrow — CHANGES_REQUESTED** [review]

**Comment by michelleangela** [comment]
Blocked for check-in until https://github.com/microsoft/vscode-cpptools/pull/3987 is merged

File: Extension/src/LanguageServer/extension.ts [hunk 4]
```diff
@@ -170,38 +171,20 @@ export async function getBuildTasks(returnComplerPath: boolean): Promise<vscode.
         }
     }
 
+    // Get known compiler paths. Do not include the known compiler path that is the same as user compiler path.
+    // Filter them based on the file type to get a reduced list appropriate for the active file.
+    let KnownCompilerPaths: string[];
```
Comment (bobbrow) [inline Extension/src/LanguageServer/extension.ts]: nit: this is probably a lint issue.  Rename: `knownCompilerPaths`

File: Extension/src/LanguageServer/extension.ts [hunk 5]
```diff
@@ -228,41 +211,56 @@ export async function getBuildTasks(returnComplerPath: boolean): Promise<vscode.
...
+        let kind: vscode.TaskDefinition = {
             type: 'shell',
             label: taskName,
             command: isCl ? compilerPathBase : compilerPath,
             args: args,
             options: isCl ? undefined : {"cwd": cwd},
-            compilerPath: isCl ? compilerPathBase : compilerPath
         };
 
+        if (returnComplerPath) {
```
Comment (bobbrow) [inline Extension/src/LanguageServer/extension.ts]: There seems to be a typo in this variable's name

**Review by bobbrow — APPROVED** [review]
Just minor issues remaining.

**Pull request merged**